docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_expiry_in_30_days_remediation.mdx

```diff
@@ -1,6 +1,261 @@
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can use the "describe-reserved-nodes" command to get information about the reserved nodes in Amazon Redshift. The command is as follows:
+
+   ```
+   aws redshift describe-reserved-nodes
+   ```
+
+3. This command will return a JSON output with details about all the reserved nodes. You need to look for the "StartTime" and "Duration" fields in the output. The "StartTime" field indicates when the reservation started, and the "Duration" field indicates the duration of the reservation in seconds.
+
+4. To find out if a reserved node lease is expiring in the next 30 days, you need to write a script that calculates the difference between the current date and the start time plus the duration. If the difference is less than or equal to 30 days, then the lease is expiring in the next 30 days. Here is a simple Python script that does this:
+
+   ```python
+   import json
+   import subprocess
+   from datetime import datetime, timedelta
+
+   # Run the AWS CLI command and get the output
+   output = subprocess.check_output(["aws", "redshift", "describe-reserved-nodes"])
+   data = json.loads(output)
+
+   # Get the current date
+   now = datetime.now()
+
+   # Check each reserved node
+   for node in data['ReservedNodes']:
+       # Get the start time and duration
+       start_time = datetime.strptime(node['StartTime'], '%Y-%m-%dT%H:%M:%S.%fZ')
+       duration = timedelta(seconds=node['Duration'])
+
+       # Calculate the expiration date
+       expiration_date = start_time + duration
+
+       # Check if the lease is expiring in the next 30 days
+       if now <= expiration_date <= now + timedelta(days=30):
+           print(f"Reserved node {node['ReservedNodeId']} is expiring in the next 30 days.")
+   ```
+   
+Please note that this script assumes that the AWS CLI is configured with access to the necessary resources. If it's not, you may need to provide the necessary credentials or configure the AWS CLI accordingly.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+2. Import the necessary libraries and establish a session with AWS using your access keys. 
+
+```python
+import boto3
+from datetime import datetime, timedelta
+
+# Create a session using your AWS credentials
+aws_session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>